react/features/filmstrip/components/web/Filmstrip.js

```diff
@@ -258,6 +229,175 @@ class Filmstrip extends Component <Props> {
         }
     }
 
+    _listItemKey: number => string;
+
+    /**
+     * The key to be used for every ThumbnailWrapper element in stage view.
+     *
+     * @param {number} index - The index of the ThumbnailWrapper instance.
+     * @returns {string} - The key.
+     */
+    _listItemKey(index) {
+        const { _remoteParticipants, _remoteParticipantsLength } = this.props;
+
+        if (typeof index !== 'number' || _remoteParticipantsLength <= index) {
+            return `empty-${index}`;
+        }
+
+        return _remoteParticipants[_remoteParticipantsLength - index - 1];
+    }
+
+    _gridItemKey: Object => string;
+
+    /**
+     * The key to be used for every ThumbnailWrapper element in tile views.
+     *
+     * @param {Object} data - An object with the indexes identifying the ThumbnailWrapper instance.
+     * @returns {string} - The key.
+     */
+    _gridItemKey({ columnIndex, rowIndex }) {
+        const { _columns, _remoteParticipants, _remoteParticipantsLength } = this.props;
+        const index = (rowIndex * _columns) + columnIndex;
+
+        if (index > _remoteParticipantsLength) {
+            return `empty-${index}`;
+        }
+
+        if (index === _remoteParticipantsLength) {
+            return 'local';
+        }
+
+        return _remoteParticipants[index];
+    }
+
+    _onListItemsRendered: Object => void;
+
+    /**
+     * Handles items rendered changes in stage view.
+     *
+     * @param {Object} data - Information about the rendered items.
+     * @returns {void}
+     */
+    _onListItemsRendered({ overscanStartIndex, overscanStopIndex }) {
+        const { dispatch, _remoteParticipants, _remoteParticipantsLength } = this.props;
+        const endIndex = Math.min(overscanStopIndex, _remoteParticipantsLength - 1);
+
+        dispatch(setVisibleParticipants(_remoteParticipants.slice(overscanStartIndex, endIndex + 1)));
+    }
+
+    _onGridItemsRendered: Object => void;
+
+    /**
+     * Handles items rendered changes in tile view.
+     *
+     * @param {Object} data - Information about the rendered items.
+     * @returns {void}
+     */
+    _onGridItemsRendered({
+        overscanColumnStartIndex,
+        overscanColumnStopIndex,
+        overscanRowStartIndex,
+        overscanRowStopIndex
+    }) {
+        const { _columns, dispatch, _remoteParticipants, _remoteParticipantsLength } = this.props;
+        const startIndex = (overscanRowStartIndex * _columns) + overscanColumnStartIndex;
+        const endIndex = Math.min(
+            (overscanRowStopIndex * _columns) + overscanColumnStopIndex,
+            _remoteParticipantsLength - 1);
+
+        dispatch(setVisibleParticipants(_remoteParticipants.slice(startIndex, endIndex + 1)));
```

Comment:
Since `onItemsRendered` is called whenever the item indices change, should we try to debounce the `setVisibleParticipants` action so that it doesn't get dispatched too many times when the user is continuously scrolling ? Same applies to stage view changes.

Reply:
I haven't noticed an issue with this currently. We will debounce the messages send to the bridge anyway. Although it might be good to do it. 

I'm a little hesitant to do it because we may need this information for rendering some stuff and then the debounce can introduce some UI glitches. 

Since I haven't noticed an issue with it currently lets leave it for later (maybe as part of the next performance optimisation PR) if needed. WDYT?